### PR TITLE
Created Model Classes And DTO S

### PR DESCRIPTION
**Pull request by qm-sowmya**

### CODE REVIEW

File: Src/Core/CaseStudy.Core/Models/Cars.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+        [Required]
+        public int DealerId { get; set; }
+
+        [Required]
+        [StringLength(255)]
+        public string Type { get; set; }
+
+        [Required]
+        [StringLength(255)]
```
Comment (Samvidsharma) [inline Src/Core/CaseStudy.Core/Models/Cars.cs]: 20 length is fine

File: Src/Core/CaseStudy.Core/Models/Cars.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    public class Cars
+    {
+        [Key]
+        public int HomeNetVehicleId { get; set; }
+
+        [Required]
+        public int DealerId { get; set; }
+
+        [Required]
+        [StringLength(255)]
```
Comment (Samvidsharma) [inline Src/Core/CaseStudy.Core/Models/Cars.cs]: 20 length is fine

File: Src/Core/CaseStudy.Core/Models/Cars.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        public string VIN { get; set; }
+
+        [Required]
+        public int Year { get; set; }
+
+        [Required]
+        [StringLength(255)]
+        public string? Make { get; set; }
```
Comment (Samvidsharma) [inline Src/Core/CaseStudy.Core/Models/Cars.cs:26]: 20 length, 
what happens if we remove ? here, is it mandatory? why?

File: Src/Core/CaseStudy.Core/Models/Components.cs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace CaseStudy.Core.Models
+{
+    public class Components
+    {
+        [Key]public int ComponentId { get; set; } 
+        public string? ComponentName { get; set; }
```
Comment (Samvidsharma) [inline Src/Core/CaseStudy.Core/Models/Components.cs]: component name should not be null

File: Src/Core/CaseStudy.Core/Models/DealerPages.cs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+﻿namespace CaseStudy.Core.Models
+{
+    public class DealerPages
+    {
+        public int Id {  get; set; }
```
Comment (Samvidsharma) [inline Src/Core/CaseStudy.Core/Models/DealerPages.cs:5]: why no attributes here?

Comment (qm-sowmya) [reply Src/Core/CaseStudy.Core/Models/DealerPages.cs:5]: As Id is automatically considered as primary while creating table, I havent mentioned explicitly

File: Src/Core/CaseStudy.Core/Models/Components.cs (hunk 4 above)
Comment (qm-sowmya) [reply Src/Core/CaseStudy.Core/Models/Components.cs]: Okay @Samvidsharma

File: Src/Core/CaseStudy.Core/Models/Cars.cs (hunk 3 above)
Comment (qm-sowmya) [reply Src/Core/CaseStudy.Core/Models/Cars.cs:26]: ? is added for handling nulls, if removed , in the operations performed in further might get exceptions

File: Src/Core/CaseStudy.Core/Models/Cars.cs (hunk 2 above)
Comment (qm-sowmya) [reply Src/Core/CaseStudy.Core/Models/Cars.cs]: okay @Samvidsharma .

**Pull request merged**